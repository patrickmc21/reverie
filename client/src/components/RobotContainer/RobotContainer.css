.App__robot-container {
  border: 1px solid #000000;
  border-radius: 10px;
  margin: auto;
  height: 620px;
  width: 90vw;
  background-color: rgba(0,0,0, 0.8);
}

.add-robot {
  float: right;
  margin: 10px 10px 0 0;
}

.card-container {
  width: 100%;
  overflow-y: scroll;
}

/*
font-family: 'Open Sans', sans-serif;
font-family: 'Roboto', sans-serif;*/