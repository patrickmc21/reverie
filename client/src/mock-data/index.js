export const mockRobots = [
  {
    id: 1,
    date_added: '5/21/2018',
    first_active: '5/21/2018',
    current_name: "R2-D2",
    height: 3.25,
    weight: 200,
    intelligence_metric: 18
  },
  {
    id: 2,
    date_added: '5/20/2018',
    first_active: '5/20/2018',
    current_name: "Bender",
    height: 6.1,
    weight: 250,
    intelligence_metric: 14
  }
];

export const newRobot = {
  date_added: '5/23/2018',
  first_active: '5/23/2018',
  current_name: "T-1000",
  height: 6.5,
  weight: 400,
  intelligence_metric: 19
};

export const updatedRobot = {
  id: 2,
  date_added: '5/23/2018',
  first_active: '5/23/2018',
  current_name: "T-1000",
  height: 6.5,
  weight: 400,
  intelligence_metric: 19
};

