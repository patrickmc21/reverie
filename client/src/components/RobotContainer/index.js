import React from 'react';
import RobotCard from '../RobotCard';
import PropTypes from 'prop-types';
import './RobotContainer.css';

const RobotContainer = ({robots, triggerForm, removeRobot}) => {
  const robotCards = robots.map((robot, index) => {
    const evenClass = index % 2 === 0 ? '' : 'even';
    return (
      <RobotCard
        key={robot.id}
        robot={robot}
        triggerForm={triggerForm}
        removeRobot={removeRobot}
        classType={evenClass}
      />
    );
  });

  return (
    <section className='App__robot-container'>
      <button 
        className='add-robot' 
        onClick={() => triggerForm()}>Add Robot</button>
      <div className='card-container'>
        <article className='robot-card'>
          <ul className='robot-card__list even'>
            <li className='robot-card__list--list-item'>
              Name:
            </li>
            <li className='robot-card__list--list-item'>
              Height (feet):
            </li>
            <li className='robot-card__list--list-item'>
              Weight (pounds):
            </li>
            <li className='robot-card__list--list-item'>
              Intelligence (1 - 20):
            </li>
            <li className='robot-card__list--list-item'>
              Date Added:
            </li>
            <li className='robot-card__list--list-item'>
              First Active:
            </li>
          </ul>
        </article>
        {robotCards}
      </div>
    </section>
  );
};

RobotContainer.propTypes = {
  robots: PropTypes.array,
  triggerForm: PropTypes.func,
  removeRobot: PropTypes.func
};

export default RobotContainer;