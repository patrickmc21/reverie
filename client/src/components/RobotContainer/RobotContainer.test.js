import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import RobotContainer from './index';
import { mockRobots } from '../../mock-data';

describe('RobotContainer', () => {

  let mockTriggerForm;
  let mockRemoveRobot;
  let wrapper;

  beforeEach(() => {
    mockTriggerForm = jest.fn();
    mockRemoveRobot = jest.fn();
    wrapper = shallow(
      <RobotContainer
        robots={mockRobots}
        triggerForm={mockTriggerForm}
        removeRobot={mockRemoveRobot} />
    );
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call triggerForm on add robot click', () => {
    wrapper.find('.add-robot').simulate('click');
    expect(mockTriggerForm).toHaveBeenCalledTimes(1);
  });
})
