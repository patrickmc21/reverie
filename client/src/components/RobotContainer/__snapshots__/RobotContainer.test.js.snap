// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RobotContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RobotContainer
    removeRobot={[Function]}
    robots={
        Array [
            Object {
              "current_name": "R2-D2",
              "date_added": "5/21/2018",
              "first_active": "5/21/2018",
              "height": 3.25,
              "id": 1,
              "intelligence_metric": 18,
              "weight": 200,
            },
            Object {
              "current_name": "Bender",
              "date_added": "5/20/2018",
              "first_active": "5/20/2018",
              "height": 6.1,
              "id": 2,
              "intelligence_metric": 14,
              "weight": 250,
            },
          ]
    }
    triggerForm={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="add-robot"
          onClick={[Function]}
>
          Add Robot
</button>,
        <div
          className="card-container"
>
          <article
                    className="robot-card"
          >
                    <ul
                              className="robot-card__list even"
                    >
                              <li
                                        className="robot-card__list--list-item"
                              >
                                        Name:
                              </li>
                              <li
                                        className="robot-card__list--list-item"
                              >
                                        Height (feet):
                              </li>
                              <li
                                        className="robot-card__list--list-item"
                              >
                                        Weight (pounds):
                              </li>
                              <li
                                        className="robot-card__list--list-item"
                              >
                                        Intelligence (1 - 20):
                              </li>
                              <li
                                        className="robot-card__list--list-item"
                              >
                                        Date Added:
                              </li>
                              <li
                                        className="robot-card__list--list-item"
                              >
                                        First Active:
                              </li>
                    </ul>
          </article>
          <RobotCard
                    classType=""
                    removeRobot={[Function]}
                    robot={
                              Object {
                                        "current_name": "R2-D2",
                                        "date_added": "5/21/2018",
                                        "first_active": "5/21/2018",
                                        "height": 3.25,
                                        "id": 1,
                                        "intelligence_metric": 18,
                                        "weight": 200,
                                      }
                    }
                    triggerForm={[Function]}
          />
          <RobotCard
                    classType="even"
                    removeRobot={[Function]}
                    robot={
                              Object {
                                        "current_name": "Bender",
                                        "date_added": "5/20/2018",
                                        "first_active": "5/20/2018",
                                        "height": 6.1,
                                        "id": 2,
                                        "intelligence_metric": 14,
                                        "weight": 250,
                                      }
                    }
                    triggerForm={[Function]}
          />
</div>,
      ],
      "className": "App__robot-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Robot",
          "className": "add-robot",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add Robot",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <article
              className="robot-card"
>
              <ul
                            className="robot-card__list even"
              >
                            <li
                                          className="robot-card__list--list-item"
                            >
                                          Name:
                            </li>
                            <li
                                          className="robot-card__list--list-item"
                            >
                                          Height (feet):
                            </li>
                            <li
                                          className="robot-card__list--list-item"
                            >
                                          Weight (pounds):
                            </li>
                            <li
                                          className="robot-card__list--list-item"
                            >
                                          Intelligence (1 - 20):
                            </li>
                            <li
                                          className="robot-card__list--list-item"
                            >
                                          Date Added:
                            </li>
                            <li
                                          className="robot-card__list--list-item"
                            >
                                          First Active:
                            </li>
              </ul>
</article>,
            Array [
              <RobotCard
                classType=""
                removeRobot={[Function]}
                robot={
                                Object {
                                                "current_name": "R2-D2",
                                                "date_added": "5/21/2018",
                                                "first_active": "5/21/2018",
                                                "height": 3.25,
                                                "id": 1,
                                                "intelligence_metric": 18,
                                                "weight": 200,
                                              }
                }
                triggerForm={[Function]}
/>,
              <RobotCard
                classType="even"
                removeRobot={[Function]}
                robot={
                                Object {
                                                "current_name": "Bender",
                                                "date_added": "5/20/2018",
                                                "first_active": "5/20/2018",
                                                "height": 6.1,
                                                "id": 2,
                                                "intelligence_metric": 14,
                                                "weight": 250,
                                              }
                }
                triggerForm={[Function]}
/>,
            ],
          ],
          "className": "card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul
                className="robot-card__list even"
>
                <li
                                className="robot-card__list--list-item"
                >
                                Name:
                </li>
                <li
                                className="robot-card__list--list-item"
                >
                                Height (feet):
                </li>
                <li
                                className="robot-card__list--list-item"
                >
                                Weight (pounds):
                </li>
                <li
                                className="robot-card__list--list-item"
                >
                                Intelligence (1 - 20):
                </li>
                <li
                                className="robot-card__list--list-item"
                >
                                Date Added:
                </li>
                <li
                                className="robot-card__list--list-item"
                >
                                First Active:
                </li>
</ul>,
              "className": "robot-card",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className="robot-card__list--list-item"
>
                    Name:
</li>,
                  <li
                    className="robot-card__list--list-item"
>
                    Height (feet):
</li>,
                  <li
                    className="robot-card__list--list-item"
>
                    Weight (pounds):
</li>,
                  <li
                    className="robot-card__list--list-item"
>
                    Intelligence (1 - 20):
</li>,
                  <li
                    className="robot-card__list--list-item"
>
                    Date Added:
</li>,
                  <li
                    className="robot-card__list--list-item"
>
                    First Active:
</li>,
                ],
                "className": "robot-card__list even",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Name:",
                    "className": "robot-card__list--list-item",
                  },
                  "ref": null,
                  "rendered": "Name:",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Height (feet):",
                    "className": "robot-card__list--list-item",
                  },
                  "ref": null,
                  "rendered": "Height (feet):",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Weight (pounds):",
                    "className": "robot-card__list--list-item",
                  },
                  "ref": null,
                  "rendered": "Weight (pounds):",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Intelligence (1 - 20):",
                    "className": "robot-card__list--list-item",
                  },
                  "ref": null,
                  "rendered": "Intelligence (1 - 20):",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Date Added:",
                    "className": "robot-card__list--list-item",
                  },
                  "ref": null,
                  "rendered": "Date Added:",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "First Active:",
                    "className": "robot-card__list--list-item",
                  },
                  "ref": null,
                  "rendered": "First Active:",
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            "type": "article",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "classType": "",
              "removeRobot": [Function],
              "robot": Object {
                "current_name": "R2-D2",
                "date_added": "5/21/2018",
                "first_active": "5/21/2018",
                "height": 3.25,
                "id": 1,
                "intelligence_metric": 18,
                "weight": 200,
              },
              "triggerForm": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "classType": "even",
              "removeRobot": [Function],
              "robot": Object {
                "current_name": "Bender",
                "date_added": "5/20/2018",
                "first_active": "5/20/2018",
                "height": 6.1,
                "id": 2,
                "intelligence_metric": 14,
                "weight": 250,
              },
              "triggerForm": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="add-robot"
            onClick={[Function]}
>
            Add Robot
</button>,
          <div
            className="card-container"
>
            <article
                        className="robot-card"
            >
                        <ul
                                    className="robot-card__list even"
                        >
                                    <li
                                                className="robot-card__list--list-item"
                                    >
                                                Name:
                                    </li>
                                    <li
                                                className="robot-card__list--list-item"
                                    >
                                                Height (feet):
                                    </li>
                                    <li
                                                className="robot-card__list--list-item"
                                    >
                                                Weight (pounds):
                                    </li>
                                    <li
                                                className="robot-card__list--list-item"
                                    >
                                                Intelligence (1 - 20):
                                    </li>
                                    <li
                                                className="robot-card__list--list-item"
                                    >
                                                Date Added:
                                    </li>
                                    <li
                                                className="robot-card__list--list-item"
                                    >
                                                First Active:
                                    </li>
                        </ul>
            </article>
            <RobotCard
                        classType=""
                        removeRobot={[Function]}
                        robot={
                                    Object {
                                                "current_name": "R2-D2",
                                                "date_added": "5/21/2018",
                                                "first_active": "5/21/2018",
                                                "height": 3.25,
                                                "id": 1,
                                                "intelligence_metric": 18,
                                                "weight": 200,
                                              }
                        }
                        triggerForm={[Function]}
            />
            <RobotCard
                        classType="even"
                        removeRobot={[Function]}
                        robot={
                                    Object {
                                                "current_name": "Bender",
                                                "date_added": "5/20/2018",
                                                "first_active": "5/20/2018",
                                                "height": 6.1,
                                                "id": 2,
                                                "intelligence_metric": 14,
                                                "weight": 250,
                                              }
                        }
                        triggerForm={[Function]}
            />
</div>,
        ],
        "className": "App__robot-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Robot",
            "className": "add-robot",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add Robot",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <article
                className="robot-card"
>
                <ul
                                className="robot-card__list even"
                >
                                <li
                                                className="robot-card__list--list-item"
                                >
                                                Name:
                                </li>
                                <li
                                                className="robot-card__list--list-item"
                                >
                                                Height (feet):
                                </li>
                                <li
                                                className="robot-card__list--list-item"
                                >
                                                Weight (pounds):
                                </li>
                                <li
                                                className="robot-card__list--list-item"
                                >
                                                Intelligence (1 - 20):
                                </li>
                                <li
                                                className="robot-card__list--list-item"
                                >
                                                Date Added:
                                </li>
                                <li
                                                className="robot-card__list--list-item"
                                >
                                                First Active:
                                </li>
                </ul>
</article>,
              Array [
                <RobotCard
                  classType=""
                  removeRobot={[Function]}
                  robot={
                                    Object {
                                                      "current_name": "R2-D2",
                                                      "date_added": "5/21/2018",
                                                      "first_active": "5/21/2018",
                                                      "height": 3.25,
                                                      "id": 1,
                                                      "intelligence_metric": 18,
                                                      "weight": 200,
                                                    }
                  }
                  triggerForm={[Function]}
/>,
                <RobotCard
                  classType="even"
                  removeRobot={[Function]}
                  robot={
                                    Object {
                                                      "current_name": "Bender",
                                                      "date_added": "5/20/2018",
                                                      "first_active": "5/20/2018",
                                                      "height": 6.1,
                                                      "id": 2,
                                                      "intelligence_metric": 14,
                                                      "weight": 250,
                                                    }
                  }
                  triggerForm={[Function]}
/>,
              ],
            ],
            "className": "card-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ul
                  className="robot-card__list even"
>
                  <li
                                    className="robot-card__list--list-item"
                  >
                                    Name:
                  </li>
                  <li
                                    className="robot-card__list--list-item"
                  >
                                    Height (feet):
                  </li>
                  <li
                                    className="robot-card__list--list-item"
                  >
                                    Weight (pounds):
                  </li>
                  <li
                                    className="robot-card__list--list-item"
                  >
                                    Intelligence (1 - 20):
                  </li>
                  <li
                                    className="robot-card__list--list-item"
                  >
                                    Date Added:
                  </li>
                  <li
                                    className="robot-card__list--list-item"
                  >
                                    First Active:
                  </li>
</ul>,
                "className": "robot-card",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <li
                      className="robot-card__list--list-item"
>
                      Name:
</li>,
                    <li
                      className="robot-card__list--list-item"
>
                      Height (feet):
</li>,
                    <li
                      className="robot-card__list--list-item"
>
                      Weight (pounds):
</li>,
                    <li
                      className="robot-card__list--list-item"
>
                      Intelligence (1 - 20):
</li>,
                    <li
                      className="robot-card__list--list-item"
>
                      Date Added:
</li>,
                    <li
                      className="robot-card__list--list-item"
>
                      First Active:
</li>,
                  ],
                  "className": "robot-card__list even",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Name:",
                      "className": "robot-card__list--list-item",
                    },
                    "ref": null,
                    "rendered": "Name:",
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Height (feet):",
                      "className": "robot-card__list--list-item",
                    },
                    "ref": null,
                    "rendered": "Height (feet):",
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Weight (pounds):",
                      "className": "robot-card__list--list-item",
                    },
                    "ref": null,
                    "rendered": "Weight (pounds):",
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Intelligence (1 - 20):",
                      "className": "robot-card__list--list-item",
                    },
                    "ref": null,
                    "rendered": "Intelligence (1 - 20):",
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Date Added:",
                      "className": "robot-card__list--list-item",
                    },
                    "ref": null,
                    "rendered": "Date Added:",
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "First Active:",
                      "className": "robot-card__list--list-item",
                    },
                    "ref": null,
                    "rendered": "First Active:",
                    "type": "li",
                  },
                ],
                "type": "ul",
              },
              "type": "article",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "classType": "",
                "removeRobot": [Function],
                "robot": Object {
                  "current_name": "R2-D2",
                  "date_added": "5/21/2018",
                  "first_active": "5/21/2018",
                  "height": 3.25,
                  "id": 1,
                  "intelligence_metric": 18,
                  "weight": 200,
                },
                "triggerForm": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "classType": "even",
                "removeRobot": [Function],
                "robot": Object {
                  "current_name": "Bender",
                  "date_added": "5/20/2018",
                  "first_active": "5/20/2018",
                  "height": 6.1,
                  "id": 2,
                  "intelligence_metric": 14,
                  "weight": 250,
                },
                "triggerForm": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
